### DISCUSSION TITLE
What is the best way to fix the model when you work with realistic data?

### DISCUSSION THREAD

**Discussion by mkondratyev85** [gempy-project/gempy#664] [General]
I've read through tutorials and watched videos on how to use `gempy`. It seems pretty straightforward when you build simple models with only a few layers and unconformities. The real struggle comes when I try to use it for building something more realistic, based on data from, say 20 boreholes.

It looks like the `gempy` often tends to construct rather surrealistic layers between formations. Briefly it was mentioned in this video tutorial https://www.youtube.com/watch?v=1oS6xTJkRwo&t=5551s. The way to deal with it - is to add more points and orientations.

It was suggested in the video to use this command:
```python
live_plot = gp.plot_3d(geo_model, plotter_type="background", show_results=True, show_lith=False)
live_plot.toggle_live_updating()
```
and then manually shift certain points and see how it changes the layers in the real time. But, again, it works fine in simple cases, but in a realistic case it's very difficult to correctly place the point in 3D space.

It would be nice if there were a way to:

1. Split the model into a set of 2D sections (vertical or tilted).
2. Manually place points or orientations in 3D space by simply clicking on those 2D sections, like if you're drawing  the contact between layers.

I know there is a way to almost solve first item  (https://docs.gempy.org/tutorials/ch1_fundamentals/ch1_3b_cross_sections.html#sphx-glr-tutorials-ch1-fundamentals-ch1-3b-cross-sections-py). But I couldn't find the way of placing set of points by clicking on the 2D plot.

Or, maybe, there is another way of fixing realistic model?

**Comment by Japhiolite** [comment]
Hi @mkondratyev85

having worked with realistic models in GemPy myself, I totally understand, where you're coming from. 
Sometimes, one or two specific interface points in the model turned out to have a strong impact on the resulting model. 

I had my go at `toggle_live_updating()` aswell, and while it helped me to clean my model, I agree that it can be tedious and chaotic with a lot of data present (wrong points get chosen due to alignment with your field of view, for instance). 

As for your suggestions: We'd need to project points onto the 2D sections. There is a threshold parameter for projecting data onto sections, but its functionality isn't well tested. Or do you mean to only show points on sections which strictly lie on a cross-section?

I have played around with bqplot in the past to pick surface points from images in cross sections (https://github.com/Japhiolite/bqcs ). Maybe it would be worth to have a look at using some libraries' interactive plotting capabilities for using them in gempy plotting...

So I don't really have a solution ready, just some thoughts...

Cheers
Jan
